eosio.system/src/voting.cpp

```diff
@@ -40,24 +40,40 @@ namespace eosiosystem {
       require_auth( producer );
 
       auto prod = _producers.find( producer );
+      const auto ct = current_time();
 
       if ( prod != _producers.end() ) {
          _producers.modify( prod, producer, [&]( producer_info& info ){
-               info.producer_key = producer_key;
-               info.is_active    = true;
-               info.url          = url;
-               info.location     = location;
+            info.producer_key = producer_key;
+            info.is_active    = true;
+            info.url          = url;
+            info.location     = location;
+            if ( info.last_claim_time == 0 )
+               info.last_claim_time = ct;
+         });
+
+         auto prod2 = _producers2.find( producer );
+         if ( prod2 == _producers2.end() ) {
+            _producers2.emplace( producer, [&]( producer_info2& info ){
+               info.owner                     = producer;
+               info.last_votepay_share_update = prod->last_claim_time;
```

Comment:
`info.last_votepay_share_update` should be set to current time, not last claim time.